### ISSUE TITLE
PreReleaseTag calculated incorrectly on master

### ISSUE THREAD

**Issue by michaelnoonan** [GitTools/GitVersion#1079]
GitVersion `4.0.0-beta.7`

We use some logic on `gitVersionInfo.PreReleaseTag` in our cake build script which decides whether to push to NuGet.org: https://github.com/OctopusDeploy/Data/blob/master/build.cake#L136

We recently changed to `gitVersionInfo.PreReleaseLabel` because the `PreReleaseTag` seems to be grabbing a string like `$".{Patch}` by accident when there is no pre-release tag.

See the following output:

```json
C:\Source\Octopus\Data>tools\GitVersion.CommandLine\tools\GitVersion.exe

{
 "Major":1,
 "Minor":0,
 "Patch":17,
 "PreReleaseTag":".17",
 "PreReleaseTagWithDash":"",
 "PreReleaseLabel":"",
 "PreReleaseNumber":"",
 "BuildMetaData":"",
 "BuildMetaDataPadded":"",
 "FullBuildMetaData":"Branch.master.Sha.c97993056704de7015020851541f7e72b93dc420",
 "MajorMinorPatch":"1.0.17",
 "SemVer":"1.0.17",
 "LegacySemVer":"1.0.17",
 "LegacySemVerPadded":"1.0.17",
 "AssemblySemVer":"1.0.17.0",
 "FullSemVer":"1.0.17",
 "InformationalVersion":"1.0.17+Branch.master.Sha.c97993056704de7015020851541f7e72b93dc420",
 "BranchName":"master",
 "Sha":"c97993056704de7015020851541f7e72b93dc420",
 "NuGetVersionV2":"1.0.17",
 "NuGetVersion":"1.0.17",
 "NuGetPreReleaseTagV2":"",
 "NuGetPreReleaseTag":"",
 "CommitsSinceVersionSource":17,
 "CommitsSinceVersionSourcePadded":"0017",
 "CommitDate":"2016-11-07"
}
```

This was a test on a branch with a pre-release tag:

```json
C:\Source\Octopus\Data>tools\GitVersion.CommandLine\tools\GitVersion.exe

{
 "Major":1,
 "Minor":0,
 "Patch":18,
 "PreReleaseTag":"gitVerTest.0",
 "PreReleaseTagWithDash":"-gitVerTest.0",
 "PreReleaseLabel":"gitVerTest",
 "PreReleaseNumber":0,
 "BuildMetaData":"",
 "BuildMetaDataPadded":"",
 "FullBuildMetaData":"Branch.gitVerTest.Sha.c97993056704de7015020851541f7e72b93dc420",
 "MajorMinorPatch":"1.0.18",
 "SemVer":"1.0.18-gitVerTest.0",
 "LegacySemVer":"1.0.18-gitVerTest0",
 "LegacySemVerPadded":"1.0.18-gitVerTest0000",
 "AssemblySemVer":"1.0.18.0",
 "FullSemVer":"1.0.18-gitVerTest.0",
 "InformationalVersion":"1.0.18-gitVerTest.0+Branch.gitVerTest.Sha.c97993056704de7015020851541f7e72b93dc420",
 "BranchName":"gitVerTest",
 "Sha":"c97993056704de7015020851541f7e72b93dc420",
 "NuGetVersionV2":"1.0.18-gitvertest0000",
 "NuGetVersion":"1.0.18-gitvertest0000",
 "NuGetPreReleaseTagV2":"gitvertest0000",
 "NuGetPreReleaseTag":"gitvertest0000",
 "CommitsSinceVersionSource":0,
 "CommitsSinceVersionSourcePadded":"0000",
 "CommitDate":"2016-11-07"
}
```

**Comment by testworksau** [comment]
We've seen this a couple of times too. 

Here's a fresh example, on my local master branch, using `GitVersion.Portable 4.0.0`

GitVersion.yml:
```
mode: Mainline
```

```
$ gitversion
{
  "Major":17,
  "Minor":1,
  "Patch":4,
  "PreReleaseTag":"",
  "PreReleaseTagWithDash":"",
  "PreReleaseLabel":"",
  "PreReleaseNumber":"",
  "BuildMetaData":"",
  "BuildMetaDataPadded":"",
  "FullBuildMetaData":"Branch.master.Sha.34510dccf9a3df07b34d4d280bd615caba4189f4",
  "MajorMinorPatch":"17.1.4",
  "SemVer":"17.1.4",
  "LegacySemVer":"17.1.4",
  "LegacySemVerPadded":"17.1.4",
  "AssemblySemVer":"17.1.4.0",
  "AssemblySemFileVer":"17.1.4.0",
  "FullSemVer":"17.1.4",
  "InformationalVersion":"17.1.4+Branch.master.Sha.34510dccf9a3df07b34d4d280bd615caba4189f4",
  "BranchName":"master",
  "Sha":"34510dccf9a3df07b34d4d280bd615caba4189f4",
  "ShortSha":"34510dc",
  "NuGetVersionV2":"17.1.4",
  "NuGetVersion":"17.1.4",
  "NuGetPreReleaseTagV2":"",
  "NuGetPreReleaseTag":"",
  "CommitsSinceVersionSource":0,
  "CommitsSinceVersionSourcePadded":"0000",
  "CommitDate":"2019-05-08"
}

$ touch newnew.py

$ git add newnew.py && git commit -m "+semver:major did something"
[master a0741b4] +semver:major did something
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 newnew.py

$ gitversion
{
  "Major":18,
  "Minor":0,
  "Patch":0,
  "PreReleaseTag":".15",
  "PreReleaseTagWithDash":"",
  "PreReleaseLabel":"",
  "PreReleaseNumber":"",
  "BuildMetaData":"",
  "BuildMetaDataPadded":"",
  "FullBuildMetaData":"Branch.master.Sha.a0741b41881d7d197a25bd89c6c2a3748519096e",
  "MajorMinorPatch":"18.0.0",
  "SemVer":"18.0.0",
  "LegacySemVer":"18.0.0",
  "LegacySemVerPadded":"18.0.0",
  "AssemblySemVer":"18.0.0.0",
  "AssemblySemFileVer":"18.0.0.0",
  "FullSemVer":"18.0.0",
  "InformationalVersion":"18.0.0+Branch.master.Sha.a0741b41881d7d197a25bd89c6c2a3748519096e",
  "BranchName":"master",
  "Sha":"a0741b41881d7d197a25bd89c6c2a3748519096e",
  "ShortSha":"a0741b4",
  "NuGetVersionV2":"18.0.0",
  "NuGetVersion":"18.0.0",
  "NuGetPreReleaseTagV2":"",
  "NuGetPreReleaseTag":"",
  "CommitsSinceVersionSource":15,
  "CommitsSinceVersionSourcePadded":"0015",
  "CommitDate":"2019-05-30"
}
```

**Comment by testworksau** [comment]
☝️ it would be great if stale bot first checked that a maintainer had acknowledged the issue, before deciding it should be closed.

**Comment by asbjornu** [comment]
@testworksau, sorry about the silence from @GitTools/developers. Is this problem reproducible with the latest version of GitVersion? If so, are you able to reproduce it with a `RepositoryFixture` test?

**Comment by asbjornu** [comment]
I also wonder whether the fix implemented in #2003 will help with this problem in any way?
